From 7d5f12ae03b37aea61a9a8cbfb2b14a47551155f Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Fri, 9 Aug 2024 12:02:39 +0200
Subject: [PATCH 1057/1695] drivers: stm32_ipcc: fix consistency checking in
 CID configuration

Fix the checking of the consistency of the CID configuration that was
only effective for the last channel of each IPCC endpoint.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 0a5d69d26de7160c6f7f0a0f38af7367022ea3ba drivers: stm32_ipcc: fix consistency checking in CID configuration]
Change-Id: I547825f23ff1fec0046fea325333750bb9089681
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/397751
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/drivers/stm32_ipcc.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/drivers/stm32_ipcc.c b/core/drivers/stm32_ipcc.c
index ca33169de..b8f98891a 100644
--- a/core/drivers/stm32_ipcc.c
+++ b/core/drivers/stm32_ipcc.c
@@ -237,7 +237,7 @@ static void apply_rif_config(struct ipcc_pdata *ipcc_d)
 		if (!is_cid_configured &&
 		    (BIT(0) & ipcc_d->conf_data->cid_confs[i])) {
 			is_cid_configured = true;
-			if (i == IPCC_NB_MAX_RIF_CHAN - 1)
+			if (i > 0)
 				panic("Inconsistent IPCC CID filtering RIF configuration");
 		}
 
@@ -267,7 +267,7 @@ static void apply_rif_config(struct ipcc_pdata *ipcc_d)
 		if (!is_cid_configured &&
 		    (BIT(0) & ipcc_d->conf_data->cid_confs[i])) {
 			is_cid_configured = true;
-			if (i == (IPCC_NB_MAX_RIF_CHAN * 2) - 1)
+			if (i > IPCC_NB_MAX_RIF_CHAN)
 				panic("Inconsistent IPCC CID filtering RIF configuration");
 		}
 
-- 
2.47.2

